### ISSUE TITLE
Column displaying relationship to user fails to load for users not Super Admin

### ISSUE THREAD

**Issue by gvocale** [strapi/strapi#13261]
### Required System information

- Node.js version: 16
- NPM version:
- Strapi version: 4.1.9
- Database: mariadb
- Operating system: Mac

### Describe the bug

For users with roles `Author` and `Editor`, opening a Collection type `Article` throws a blank page after few seconds, with console log error:
```tsx
TypeError: Cannot read properties of undefined (reading 'count')
```

This error happens only to `Author` and `Editor` role users. A user with `Super Admin` role is able to open the page correctly.

In our case, the collection `Article` has a field `authors`, which is a many to many relationship to `users`. I suspect somehow non Super Admin users are missing the authorization to display authors in a column.

Removing the column `authors` from `Configure the view` fixed the issue for `Author` and `Editor` users.

### Steps to reproduce the behavior

1. Go to a Collection type page, i.e. /admin/content-manager/collectionType/api::article.article?page=1&pageSize=10&sort=title:ASC
2. Choose `Configure the view`
3. Choose a field that is a many to many relationship between `users` and `articles`. In our case this field is called `authors`
4. Log out
5. Log in as user that has roles `Author` and `Editor`
1. Open /admin/content-manager/collectionType/api::article.article?page=1&pageSize=10&sort=title:ASC
6. After a second, the page is white and the Chrome console show error TypeError: Cannot read properties of undefined (reading 'count')

### Expected behavior

The page should load correctly.

### Screenshots

<img width="572" alt="MicrosoftTeams-image (30)" src="https://user-images.githubusercontent.com/1319291/167211301-1d18dd41-1878-4b3b-ba05-591205879b2e.png">

![MicrosoftTeams-image (31)](https://user-images.githubusercontent.com/1319291/167211337-e79944e7-76c9-42ec-a9c2-bd4640f7ff2c.png)

<img width="1912" alt="Screen Shot 2022-05-06 at 4 22 59 PM" src="https://user-images.githubusercontent.com/1319291/167211792-dfeb329b-070f-45fb-80aa-2e41d50e54ae.png">

<img width="1900" alt="Screen Shot 2022-05-06 at 4 26 51 PM" src="https://user-images.githubusercontent.com/1319291/167212035-758b8256-db4d-45f3-897a-d37fce91165b.png">

<img width="1993" alt="Screen Shot 2022-05-06 at 4 29 09 PM" src="https://user-images.githubusercontent.com/1319291/167212178-40ce736e-c96d-4414-b1a8-321ee6d9212a.png">

```tsx
// article/schema.json
    "authors": {
      "type": "relation",
      "relation": "manyToMany",
      "target": "plugin::users-permissions.user",
      "inversedBy": "articles"
    }
```

**Comment by gvocale** [comment]
To add to it....`Editor` was set to not have rights to query for `User`, which I guess would contribute to causing the error...although a more graceful fallback could be desired 😄 

<img width="2002" alt="Screen Shot 2022-05-06 at 4 50 57 PM" src="https://user-images.githubusercontent.com/1319291/167214834-f24246da-3369-45b6-9c1d-ba6dbd8a16ac.png">

**Comment by dqvn** [comment]
Not a technical bug but cms operation issue. For my case, same console outputs, I open administrator UI > Roles > Editor configuration > check all fields inside the entities that having this error > save 
Then the Editor user can view the page without error in console log!

**Comment by gvocale** [comment]
@alexandrebodin maybe some sort of anonymous avatar profile indicating how many private users are related to that entry?

For example facebook shows how many private users have clicked "Going" to a party like this:
<img width="560" alt="Screen Shot 2022-06-01 at 8 19 13 PM" src="https://user-images.githubusercontent.com/1319291/171522574-3852d6c9-8501-4f93-a3ea-506a3f8716ae.png">

**Comment by alexandrebodin** [comment]
Thank you for sharing this idea :) I'll let our content squad review the issue and see how they want to handle it :)

**Comment by ronronscelestes** [comment]
I still was not able to reproduce this issue on 4.2.2 - 4.2.3 or 4.3.0.
It was likely fixed by #13705.
Don't hesitate to reopen it if you still encounter this problem 😊
